stdlib/public/RemoteInspection/TypeLowering.cpp

```diff
@@ -2196,6 +2196,34 @@ class LowerType
   TypeConverter &TC;
   remote::TypeInfoProvider *ExternalTypeInfo;
 
+  const TypeInfo *CFRefTypeInfo(const TypeRef *TR) {
+    if (auto N = dyn_cast<NominalTypeRef>(TR)) {
+      Demangler Dem;
+      auto Node = N->getDemangling(Dem);
+      if (Node->getKind() == Node::Kind::Type && Node->getNumChildren() == 1) {
+	auto Alias = Node->getChild(0);
+	if (Alias->getKind() == Node::Kind::TypeAlias && Alias->getNumChildren() == 2) {
+	  auto Module = Alias->getChild(0);
+	  auto Name = Alias->getChild(1);
+	  if (Module->getKind() == Node::Kind::Module
+	      && Module->hasText()
+	      && Module->getText() == "__C"
+	      && Name->getKind() == Node::Kind::Identifier
+	      && Name->hasText()) {
+	    auto CName = Name->getText();
+	    // Heuristic: Hopefully good enough.
```

Comment:
It would be nice if we could go out and find a matching foreign type descriptor to get a definitive answer about what the type is. But that would be a lot more difficult, and this heuristic should definitely be good enough. We can keep that in our pocket in case this doesn't work right somewhere.